src/common/libsubprocess/subprocess.c

```diff
@@ -794,6 +795,24 @@ static int subprocess_read (flux_subprocess_t *p,
         return -1;
     }
 
+    if (p->flags & FLUX_SUBPROCESS_FLAGS_LOCAL_UNBUF) {
+        if (!local_unbuf_allowed) {
+            errno = EPERM;
+            return -1;
+        }
+        /* If user called this function twice, return nothing.  They
+         * will hopefully call flux_subprocess_read_stream_closed() to
+         * verify EOF was not yet reached.
+         */
+        if (c->unbuf_consumed)
+            return 0;
```

Comment:
The flag and the check seem unnecessary.  We can probably just document that reading twice returns the same data and it won't be very surprising given the lack of a buffer.

See doc comment below.

Reply:
> The flag and the check seem unnecessary. We can probably just document that reading twice returns the same data and it won't be very surprising given the lack of a buffer.

I was trying to be consistent to how the buffered case works.  But yeah, we could tweak that.